web/src/message_list_view.js

```diff
@@ -223,7 +238,8 @@ function populate_group_from_message_container(group, message_container) {
         group.all_visibility_policies = user_topics.all_visibility_policies;
     } else if (group.is_private) {
         group.pm_with_url = message_container.pm_with_url;
-        group.display_reply_to = message_store.get_pm_full_names(message_container.msg);
```

Comment:
I added new function instead of using `get_pm_full_names` because the string returned by `get_pm_full_names` and `get_display_full_name`, which is used by `get_pm_full_names` to get the user names from IDs, is used as text and not html in many places. So changing the code in `get_display_full_name` to have '<i>` tag around "guest" results in the text being shown as `Polonius <i>(guest)</i>`.

We can probably fix those while being careful about escaping and translation but would be better to do that in a separate PR.

Reply:
Good catch, @sahil839. One thing to note is that pills and sidebars that use flexbox are often having to work with usernames as plain text nodes. So an even better solution might be to have an HTML element surrounding both (e.g., `<span class="some-new-unique-user-class">Polonius <i>(guest)</i></span>`). That would likely work better with any existing flexboxes (which create an anonymous box around any text nodes), and prevent the `<i>` portion from becoming its own flex item, which could make for a whole lot of weird bugs on flexboxes alone.

Reply:
@karlstolley  So, if I am understanding correctly, this means that we should make sure in cases where flexbox layout is used the name and "<i>(guest)</i>" element should be inside a single HTML element if it is direct children of a flex container.

Reply:
Yes. And you'd have to set it up so that the `text-overflow` properties were on that `.some-new-unique-user-class`, because it would be the username text-node that would get the ellipsis, while keeping the `<i>(guest)</i>` portion always visible. It might take some tinkering to get that to work as expected inside existing flex containers, but I'm always happy to jam on this kind of problem if you think I could be of some help.